src/prefect/server/api/variables.py

```diff
@@ -50,9 +51,15 @@ async def create_variable(
     db: PrefectDBInterface = Depends(provide_database_interface),
 ) -> core.Variable:
     async with db.session_context(begin_transaction=True) as session:
-        model = await models.variables.create_variable(
-            session=session, variable=variable
-        )
+        try:
+            model = await models.variables.create_variable(
+                session=session, variable=variable
+            )
+        except sa.exc.IntegrityError:
+            raise HTTPException(
+                status_code=409,
+                detail=f"A variable with the name {variable.name!r} already exists.",
+            )
```

Comment:
This API change ensures a useful error is returned when a variable with the same name already exists.